datafusion/core/tests/parquet_page_index_pruning.rs

```diff
@@ -0,0 +1,205 @@
+    //  vec.push(RowSelector::skip(3330));
+    //  vec.push(RowSelector::select(312));
+    //  vec.push(RowSelector::skip(2430));
+    //  combine with before filter total 1275 row
+    assert_eq!(batch.num_rows(), 1275);
+
+    // 3. create filter month == 1 and month == 12;
+    let filter = col("month")
+        .eq(lit(1_i32))
+        .and(col("month").eq(lit(12_i32)));
+
+    let parquet_exec = get_parquet_exec(filter, session_ctx.clone()).await;
+
+    let mut results = parquet_exec.execute(0, task_ctx.clone()).unwrap();
+
+    let batch = results.next().await;
+
+    assert!(batch.is_none());
+
+    // 4.create filter 0 < month < 2 ;
+    let filter = col("month").gt(lit(0_i32)).and(col("month").lt(lit(2_i32)));
+
+    let parquet_exec = get_parquet_exec(filter, session_ctx.clone()).await;
+
+    let mut results = parquet_exec.execute(0, task_ctx.clone()).unwrap();
+
+    let batch = results.next().await.unwrap().unwrap();
+
+    // should same with `month = 1`
+    assert_eq!(batch.num_rows(), 645);
+}
+
+#[tokio::test]
+async fn page_index_filter_multi_col() {
+    let session_ctx = SessionContext::new();
+    let task_ctx = session_ctx.task_ctx();
+
+    // create filter month == 1 and year = 2009;
+    let filter = col("month").eq(lit(1_i32)).and(col("year").eq(lit(2009)));
+
+    let parquet_exec = get_parquet_exec(filter, session_ctx.clone()).await;
+
+    let mut results = parquet_exec.execute(0, task_ctx.clone()).unwrap();
+
+    let batch = results.next().await.unwrap().unwrap();
+
+    //  `year = 2009` from the page index should create below RowSelection
+    //  vec.push(RowSelector::select(3663));
+    //  vec.push(RowSelector::skip(3642));
+    //  combine with `month = 1` total 333 row
+    assert_eq!(batch.num_rows(), 333);
+
+    // create filter (year = 2009 or id = 1) and month = 1;
+    // this should only use `month = 1` to evaluate the page index.
+    let filter = col("month")
+        .eq(lit(1_i32))
+        .and(col("year").eq(lit(2009)).or(col("id").eq(lit(1))));
+
+    let parquet_exec = get_parquet_exec(filter, session_ctx.clone()).await;
+
+    let mut results = parquet_exec.execute(0, task_ctx.clone()).unwrap();
+
+    let batch = results.next().await.unwrap().unwrap();
+    assert_eq!(batch.num_rows(), 645);
+
+    // create filter (year = 2009 or id = 1)
+    // this filter use two columns will not push down
+    let filter = col("year").eq(lit(2009)).or(col("id").eq(lit(1)));
+
+    let parquet_exec = get_parquet_exec(filter, session_ctx.clone()).await;
+
+    let mut results = parquet_exec.execute(0, task_ctx.clone()).unwrap();
+
+    let batch = results.next().await.unwrap().unwrap();
+    assert_eq!(batch.num_rows(), 7300);
+
+    // create filter (year = 2009 and id = 1) or (year = 2010)
+    // this filter use two columns will not push down
+    // todo but after use CNF rewrite it could rewrite to (year = 2009 or  year = 2010) and (id = 1 or year = 2010)
```

Comment:
Not sure about the CNF rewrite comment here

Reply:
Maybe we file a ticket do some tests.

Reply:
I am not sure what tests you have in mind here so I did not file a ticket -- perhaps you could add to one of the existing tickets (https://github.com/apache/arrow-datafusion/issues/4087 ?) or file a new one?